aten/src/ATen/core/jit_type.h

```diff
@@ -324,25 +324,60 @@ inline c10::optional<T> merge_primitive(
   return c10::optional<T>{};
 }
 
-// `VaryingShape` tracks if individual dimensions or a rank vary across
-// profiled runs. A *varying* or *dynamic* dimension is expressed as
-// an empty c10::optional in `sizes_`. If a rank is dynamic, the entire
-// `sizes_` becomes the empty optional.
+struct CAFFE2_API ShapeSymbol {
+    
+    
+    // needed for use in `std::map`
+    ShapeSymbol(): value_(-1) {}
+    bool is_static() const { return value_ >= 0; };  
```

Comment:
what does is_static mean? this could use a comment.

Reply:
What does 'fixed' mean? (I can guess but developers new to the codebase could use more specific wording).